Tutorial chapter about configuring FastDDS middleware with XML files

IkerLuengo opened the pull request:
Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

maryaB-osr committed:
Versioned Rolling branch (#907)

* committing version test

* remove tags

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* adding whitelists and trying released/in-development

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* comment out unused config with wrong type causing warning

* Enable pulling from origin and released flag

* Example warning banner

* reverse order and move version list below navigation

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* reorder version list

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* refactored after separating versions

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* lowercase branch names, add dashing

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* fixing branch versioning

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* travis.yml: pip3 install sphinx_multiversion

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* remove missing-reference function

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* fix typo

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* no env config video for rolling

Signed-off-by: maryaB-osr <marya@openrobotics.org>

Co-authored-by: Tully Foote <tfoote@osrfoundation.org>

clalancette committed:
Add in a Dockerfile to build the multiversion.

This will be used by the buildfarm job to build the documentation
and upload it to docs.ros.org.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

maryaB-osr committed:
test label-based backport with mergify (#939)

Signed-off-by: maryaB-osr <marya@openrobotics.org>

maryaB-osr committed:
test page-based backport with mergify (#943)

Signed-off-by: maryaB-osr <marya@openrobotics.org>

mergify[bot] committed:
XML launch files are still supported in ROS2 (#947) (#954)

* XML launch files are still supported in ROS2

Should this also mention YAML? I saw in the source code that there is also a YAML frontend.

* Shorten long sentence

* Add "and YAML"

(cherry picked from commit e86c950f6dfb54841ac56e840fb3f4a4e49c4e7a)

Co-authored-by: Nikolai Morin <nnmmgit@gmail.com>

mergify[bot] committed:
Fix a small typo in the Discovery Server documentation. (#937) (#955)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 34b1b3b63ca93f3f38da0314e1334b46c073ead5)

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
Fix c++ example spelling (#951) (#969)

(cherry picked from commit 69712f718e3ae562b52b442f6f8d5988d68c445d)

Co-authored-by: Andre Nguyen <andre-nguyen@users.noreply.github.com>

mergify[bot] committed:
Update the ROS 2 roadmap. (#964) (#972)

Mark items as complete, add in new items, and update time
estimates.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit aa228e9bfdbde7b566e6434c6edf0b88a1cc717b)

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
Adding ROS World links to ROSCon Content page (#952) (#962)

Signed-off-by: maryaB-osr <marya@openrobotics.org>
(cherry picked from commit d73b0303a0100caa1bceeb65b27e3e52a0fc3c1d)

Co-authored-by: Marya Belanger <marya@openrobotics.org>

mergify[bot] committed:
Update Writing-A-Simple-Cpp-Publisher-And-Subscriber.rst (#965) (#982)

(cherry picked from commit 6c170abcc04890bdea76f57f713ce4cb8da8df11)

Co-authored-by: woodz- <woodz@andechs.net>

mergify[bot] committed:
Switch from Travis -> Github actions. (#948) (#953)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit dab709012c56be87d795efc5a229714877ec07ee
and commit 9289f218944de099158c65e0a6bcf699d804ef35)

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
Add a section on becoming a core maintainer (#992) (#993)

* Add a section on becoming a core maintainer.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
Co-authored-by: William Woodall <william+github@osrfoundation.org>
(cherry picked from commit 654c0c2c4f751fd467057e3cf196ab372a8745c0)

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Update the tutorial to point to Cyclone DDS as the default.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 1edebe38b52a728b608bf7711f1bfe33533e9131)

# Conflicts:
#	source/Tutorials/Working-with-multiple-RMW-implementations.rst

clalancette committed:
Add a release note about changing the default RMW to Cyclone DDS.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 22800cdb650215eff5993ef3baf8c339f56acc52)

clalancette committed:
Update Concepts document to point to Cyclone DDS as default.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit d944d4d89bb731c8274aa5e937e7d0c40bb5d901)

clalancette committed:
Update Rolling installation instructions to mention Cyclone DDS.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 92710393000e35b10fff5e3839fb73879fdcd202)

maryaB-osr committed:
Add EOL banner to eloquent (#990)

* Add EOL banner to eloquent

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* Change the wording for version banners (#991)

* Change the wording for version banners

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* Apply suggestions from code review

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

* add commas to review change

Signed-off-by: maryaB-osr <marya@openrobotics.org>

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
give an idea to build the msg interface later (#979) (#997)

* Update Single-Package-Define-And-Use-Interface.rst

* Update Single-Package-Define-And-Use-Interface.rst

fix typo

* Update source/Tutorials/Single-Package-Define-And-Use-Interface.rst

Co-authored-by: Marya Belanger <marya@openrobotics.org>
(cherry picked from commit afae75ec4d0bc07d7934bb4f51ac510211d0ec3d)

Co-authored-by: woodz- <woodz@andechs.net>

clalancette committed:
Fix conflicts.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
add links anywhere "previous tutorial" is mentioned (#1001) (#1005)

Signed-off-by: maryaB-osr <marya@openrobotics.org>
(cherry picked from commit a34a06747e46a34ee682ec491055fd6ffb956b03)

Co-authored-by: Marya Belanger <marya@openrobotics.org>

mergify[bot] committed:
Sign up AWS RoboMaker for rosbag2 /clock roadmap item (#1011) (#1012)

(cherry picked from commit b0f30171795255fc12a8a850a923eddb9f8e69f3)

Co-authored-by: Emerson Knapp <537409+emersonknapp@users.noreply.github.com>

mergify[bot] committed:
Add Rust Working Group (#893) (#1015)

Signed-off-by: ruffsl <roxfoxpox@gmail.com>
(cherry picked from commit 438bc794f049bdf8e79ae3fce780503cfc689633)

Co-authored-by: Ruffin <roxfoxpox@gmail.com>

mergify[bot] committed:
Correct package names (#1024) (#1025)

(cherry picked from commit e730d337a8cc5138d648294265cbdf8a3dbbb56e)

Co-authored-by: Geoffrey Biggs <gbiggs@killbots.net>

mergify[bot] committed:
added core content to concepts, renamed files to match (#1010) (#1029)

Signed-off-by: maryaB-osr <marya@openrobotics.org>
(cherry picked from commit 109b41a4be1cca78d269ee38e020bbc53699a289)

Co-authored-by: Marya Belanger <marya@openrobotics.org>

mergify[bot] committed:
Add iRobot and Sony. (#1034) (#1035)

Co-authored-by: Brian Gerkey <gerkey@openrobotics.org>
(cherry picked from commit 7fd6133625ae8d3d8ea075d366fec475bafa09ea)

Co-authored-by: gerkey <gerkey@osrfoundation.org>

mergify[bot] committed:
Add Galactic feature note for externally configuring QoS (#1033) (#1039)

* Add Galactic feature note for externally configuring QoS

Signed-off-by: Jacob Perron <jacob@openrobotics.org>
(cherry picked from commit 76c31dcc7faa646a49446da0fca730c272f4f6ad)

Co-authored-by: Jacob Perron <jacob@openrobotics.org>

clalancette committed:
Update Makefile and requirements.txt.

In particular, make it so that requirements.txt has all of
the requirements (including sphinx-multiversion), and that the
Makefile has a target for building multiversion.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
Remove Sony logo pending permission for use in this context (#1044) (#1045)

* Remove Sony logo pending permission for use in this context

Co-authored-by: Brian Gerkey <gerkey@openrobotics.org>
(cherry picked from commit 49c28d888cf97a2c8cfbcfa406e3f84b47696cc9)

Co-authored-by: gerkey <gerkey@osrfoundation.org>

mergify[bot] committed:
Small typo in Understanding-ROS2-Topics.rst (#1048) (#1049)

(cherry picked from commit 0bfc16718f59bbcf06ef21e98b9e860ae4f6dc4c)

Co-authored-by: Abrar Rahman Protyasha <aprotyas@u.rochester.edu>

mergify[bot] committed:
Add navigation and SLAM demos (in simulation) (#1022) (#1052)

(cherry picked from commit f97722210c332f174dff6eed0156d2c9eaa3c739)

Co-authored-by: Darko Lukić <lukicdarkoo@gmail.com>

mergify[bot] committed:
Update the roadmap with the current status. (#1043) (#1055)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 823189c28846208202ebb813f732559a1f1db624)

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
Improve visual presentation of TSC member logos. (#1058) (#1059)

Change from `image::` to `figure::`, which automatically pads around
images. Also move the image includes directly into the table, where
they're easier to edit. And tweak sizes on a few logos to make them
appear more equally sized.

Co-authored-by: Brian Gerkey <gerkey@openrobotics.org>
(cherry picked from commit 0cfb0d40da6b0202aeb6425f770c302e92d9c52d)

Co-authored-by: gerkey <gerkey@osrfoundation.org>

mergify[bot] committed:
fixed glob command to correctly find launch files (#1042) (#1062)

* fixed glob command to correctly find launch files

tested on nvidia jetson xavier, ubuntu 18.04, ros2 eloquent

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
Co-authored-by: Chris Lalancette <clalancette@gmail.com>
(cherry picked from commit 5163254c6a02bf55b9732ecc09b5229b11be291e)

Co-authored-by: PhilippPolterauer <45892981+PhilippPolterauer@users.noreply.github.com>

mergify[bot] committed:
Add Sony logo (now with permission). (#1065) (#1067)

Also fix alphabetical ordering by moving GVSC (its prior position was a
holdover of the org's old name, TARDEC).

(cherry picked from commit ada0eb80754537a83fc7c253a0a6b9374c324ae5)

Co-authored-by: gerkey <gerkey@osrfoundation.org>

mergify[bot] committed:
Update Understanding-ROS2-Services.rst (#1073) (#1074)

The correct command line is ros2 interface show turtlesim/srv/Spawn
The original command line ros2 interface show turtlesim/srv/Spawn.srv results in an error.
:-)

(cherry picked from commit 25f723c7b4a05231bcd1d1aa99a827e797ef780d)

Co-authored-by: James Kitching <JamesKitching@users.noreply.github.com>

mergify[bot] committed:
Updated Using-Parameters-In-A-Class-Python.rst (bp #1070) (#1077)

* Updated Using-Parameters-In-A-Class-Python.rst (#1070)

* Updated Using-Parameters-In-A-Class-Python.rst

* Updated Using-Parameters-In-A-Class-Python.rst

* Updated Using-Parameters-In-A-Class-Python.rst

* Updated formatting

* Updated formatting

* Updated formatting

* Update source/Tutorials/Using-Parameters-In-A-Class-Python.rst

Co-authored-by: Marya Belanger <marya@openrobotics.org>
(cherry picked from commit f09b05fd4adb2d4d7d210a8c269a769d67b2827a)

# Conflicts:
#	source/Tutorials/Using-Parameters-In-A-Class-Python.rst

* fix merge conflicts

Signed-off-by: maryaB-osr <marya@openrobotics.org>

Co-authored-by: Archit Rungta <rungtarchit1@gmail.com>
Co-authored-by: maryaB-osr <marya@openrobotics.org>

IkerLuengo committed:
Tutorial chapter about configuring FastDDS middleware with XML files

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

mergify[bot] committed:
Amend charter with 3 changes following TSC vote in Feb 2021 (#1086) (#1087)

(cherry picked from commit 6ec07cb1369b067aec9f708b85fbb08cce214c72)

Co-authored-by: gerkey <gerkey@osrfoundation.org>

clalancette committed:
Update the canonical URL to point to the correct place. (#1082)

* Update the canonical URL to point to the correct place.

Unfortunately, in a world where we are using sphinx multiversion,
this isn't quite enough since we can't embed the current distribution
name into the config.  Also add in a hook that will rewrite the
html_baseurl right after the configuration has been finished
initializing.

While we are in here, also update a couple of URLs and dates.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add a redirect from the directory page to the latest index.html

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Make sure to sort the branches properly for the sidebar.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add in a banner for discussions, status, and Q&A.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add in "Edit on Github" links.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Fix github workflow to use requirements.txt.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+
+Prerequisites¶
+-------------
```
```suggestion
Prerequisites
-------------
```

This character also causes a warning when building the documentation

```
/home/eprosima/ROS/ros2_documentation/source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:34: WARNING: Title underline too short.

Prerequisites¶
-------------
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Background
+----------
+
+The interface between the ROS 2 stack and *Fast DDS* is provided by a ROS 2 package `rmw_fastrtps <https://github.com/ros2/rmw_fastrtps>`_.
```
```suggestion
The interface between the ROS 2 stack and *Fast DDS* is provided by the ROS 2 middleware implementation `rmw_fastrtps <https://github.com/ros2/rmw_fastrtps>`_.
```

Technically speaking there are three ROS 2 packages related with the linked repository:
* `rmw_fastrtps_shared_cpp`
* `rmw_fastrtps_cpp`
* `rmw_fastrtps_dynamic_cpp`

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+----------
+
+The interface between the ROS 2 stack and *Fast DDS* is provided by a ROS 2 package `rmw_fastrtps <https://github.com/ros2/rmw_fastrtps>`_.
+This package is available in all ROS 2 distributions, both from binaries and from sources.
```
```suggestion
This implementation is available in all ROS 2 distributions, both from binaries and from sources.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+-------------
+
+This tutorial assumes you have a ROS 2 Foxy (or newer) installation, and that you know how to :ref:`create a package <CreatePkg>`.
+It also assumes you know how to write a :ref:`simple publisher and subscriber<CppPubSub>` and a `simple server and client<CppSrvCli>`.
```
```suggestion
It also assumes you know how to write a :ref:`simple publisher and subscriber<CppPubSub>` and a :ref:`simple service and client<CppSrvCli>`.
```

The linked ROS 2 tutorial uses the term service instead of server (even though the example names the executable server).

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+This tutorial assumes you have a ROS 2 Foxy (or newer) installation, and that you know how to :ref:`create a package <CreatePkg>`.
+It also assumes you know how to write a :ref:`simple publisher and subscriber<CppPubSub>` and a `simple server and client<CppSrvCli>`.
+Although The examples are implemented on C++, the same concepts apply to Python packages.
```
```suggestion
Although the examples are implemented on C++, the same concepts apply to Python packages.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Mixing synchronous and asynchronous publications in the same node
+-----------------------------------------------------------------
+
+On this first example, we will be creating a node with two publishers, one of them with synchronous publication mode and the other one with asynchronous publication mode.
```
```suggestion
On this first example, a node with two publishers, one of them with synchronous publication mode and the other one with asynchronous publication mode, will be created.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Create the node with the publishers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Start creating the node with both publishers. Add new source file named ``sync_async_writer.cpp`` on your package with the following content:
```
```suggestion
Start creating the node with both publishers.
First, create a new package called ``sync_async_node_example_cpp``.
Add a new source file named ``sync_async_writer.cpp`` on your package with the following content:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+
+Note that the synchronous publisher will be publishing on topic ``sync_topic``, while the asynchronous one will be publishing on topic ``async_topic``.
+Add a new target on the ``CMakeLists.txt`` file of your package with this code and name it ``SyncAsyncWriter``, so that you can run the node using ``ros2 run``.
```
```suggestion

Add a new target on the ``CMakeLists.txt`` file of your package with this code and name it ``SyncAsyncWriter``, so that you can run the node using ``ros2 run``.
```

I suggest starting a new paragraph

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Note that the synchronous publisher will be publishing on topic ``sync_topic``, while the asynchronous one will be publishing on topic ``async_topic``.
+Add a new target on the ``CMakeLists.txt`` file of your package with this code and name it ``SyncAsyncWriter``, so that you can run the node using ``ros2 run``.
+If you build and run this node now, both publishers will behave the same. We still need to apply the configuration for the publication mode.
+However, this configuration will be applied on runtime suring the launch of the node, using an XML file.
```
```suggestion
If this node is built and run now, both publishers will behave the same, publishing asynchronously in both topics, because this is the default publication mode.
The default publication mode configuration can be changed on runtime during the node launching, using an XML file.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+     </profiles>
+
+Not that we have a publisher profile for ``sync_topic``, and another one for ``async_topic``.
```
```suggestion
Note that several publisher profiles are defined.
One default profile which is defined setting the ``is_default_profile`` to true, and two profiles which names coincide with those of the previously defined topics: ``sync_topic`` and ``async_topic``.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+     </profiles>
+
+Not that we have a publisher profile for ``sync_topic``, and another one for ``async_topic``.
+Each of these profiles set the publication mode to ``SYNCHRONOUS`` or ``ASYNCHRONOUS`` accordingly.
```
```suggestion
These last two profiles set the publication mode to ``SYNCHRONOUS`` or ``ASYNCHRONOUS`` accordingly.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Not that we have a publisher profile for ``sync_topic``, and another one for ``async_topic``.
+Each of these profiles set the publication mode to ``SYNCHRONOUS`` or ``ASYNCHRONOUS`` accordingly.
+Note also that there is a default profile, and that all profiles specify a ``historyMemoryPolicy`` value.
+These are needed for the example to work, the reason will be explained later on this tutorial.
```
```suggestion
Note also that all profiles specify a ``historyMemoryPolicy`` value which is needed for the example to work, and the reason will be explained later on this tutorial.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+      export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      export RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      export FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      export FASTRTPS_DEFAULT_PROFILES_FILE=path/to/SyncAsync.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+      export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      export RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      export FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      export FASTRTPS_DEFAULT_PROFILES_FILE=path/to/SyncAsync.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+      SET RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      SET RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      SET FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      SET FASTRTPS_DEFAULT_PROFILES_FILE=path/to/SyncAsync.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+    [INFO] [1612972050.494478706] [minimal_publisher]: Synchronously publishing: 'SYNC: Hello, world! 1'
+    [INFO] [1612972050.494664334] [minimal_publisher]: Asynchronously publishing: 'ASYNC: Hello, world! 1'
+    [INFO] [1612972050.994368474] [minimal_publisher]: Synchronously publishing: 'SYNC: Hello, world! 2'
+    [INFO] [1612972050.994549851] [minimal_publisher]: Asynchronously publishing: 'ASYNC: Hello, world! 2'
```
```suggestion
    [INFO] [1612972049.994630332] [sync_async_publisher]: Synchronously publishing: 'SYNC: Hello, world! 0'
    [INFO] [1612972049.995097767] [sync_async_publisher]: Asynchronously publishing: 'ASYNC: Hello, world! 0'
    [INFO] [1612972050.494478706] [sync_async_publisher]: Synchronously publishing: 'SYNC: Hello, world! 1'
    [INFO] [1612972050.494664334] [sync_async_publisher]: Asynchronously publishing: 'ASYNC: Hello, world! 1'
    [INFO] [1612972050.994368474] [sync_async_publisher]: Synchronously publishing: 'SYNC: Hello, world! 2'
    [INFO] [1612972050.994549851] [sync_async_publisher]: Asynchronously publishing: 'ASYNC: Hello, world! 2'
```

To be consistent with the node name in the source code.

JLBuenoLopez reviewed (commented):
Partial review

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+.. code-block:: console
+
+    ros2 run SyncAsyncWriter
```
```suggestion
    ros2 run  sync_async_node_example_cpp SyncAsyncWriter
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+    ros2 run SyncAsyncWriter
+
+You should see the subscribers receiving the data from the publishing node, like so:
```
```suggestion
You should see the publishers sending the data from the publishing node, like so:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Create a node with the subscribers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can create a new node with the subscribers that will listen to the ``sync_topic`` and ``async_topic`` publications.
```
```suggestion
Next, a new node with the subscribers that will listen to the ``sync_topic`` and ``async_topic`` publications is going to be created.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Add the subscriber profiles to the the XML
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can add configuration profiles for the subscribers in the same XML file.
```
```suggestion
Configuration profiles for the subscribers can be added in the same XML file.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can add configuration profiles for the subscribers in the same XML file.
+For the moment, we will set only a default profile. We will extend the subscriber profiles later on.
```
```suggestion
For the moment, only the default profile will be set. The subscriber profiles will be extended later on.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+      export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      export RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      export FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      export FASTRTPS_DEFAULT_PROFILES_FILE=path/to/SyncAsync.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+      export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      export RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      export FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      export FASTRTPS_DEFAULT_PROFILES_FILE=path/to/SyncAsync.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+      SET RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      SET RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      SET FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      SET FASTRTPS_DEFAULT_PROFILES_FILE=path/to/SyncAsync.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+.. code-block:: console
+
+    ros2 run SyncAsyncReader
```
```suggestion
    ros2 run sync_async_node_example_cpp SyncAsyncReader
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+The only requirement is that all publisher profiles have a different name and that there is only a single default profile.
+The same goes for subscribers.
+
+In order to define a configuration for a specific topic, just make the name of the profile to be the name of the ROS2 topic (like ``/sync_topic`` and ``/async_topic`` in the example),
```
```suggestion
In order to define a configuration for a specific topic, just name the profile after the ROS 2 topic name (like ``/sync_topic`` and ``/async_topic`` in the example),
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+The same goes for subscribers.
+
+In order to define a configuration for a specific topic, just make the name of the profile to be the name of the ROS2 topic (like ``/sync_topic`` and ``/async_topic`` in the example),
+and ``rmw_fastrtps`` will apply the profile to all publishers and subscribers for that topic.
```
```suggestion
and ``rmw_fastrtps`` will apply this profile to all publishers and subscribers for that topic.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+and ``rmw_fastrtps`` will apply the profile to all publishers and subscribers for that topic.
+The default configuration profile is identified by the attribute ``is_default_profile`` set to ``true``, and acts as a fallback profile when there is no other one with a name matching the topic name.
+
+The environment variable ``FASTRTPS_DEFAULT_PROFILES_FILE`` is used to inform ``rmw_fastrtps`` the path to the XML file with the configuration profiles to load.
```
```suggestion
The environment variable ``FASTRTPS_DEFAULT_PROFILES_FILE`` is used to inform *Fast DDS* of the path to the XML file with the configuration profiles to load.
```

This environment variable is technically a *Fast DDS* variable and can be used outside the ROS 2 stack scope.

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Among all the configurable attributes, ``rmw_fastrtps`` treats ``publishMode`` and ``historyMemoryPolicy`` differently.
+By default, these values are set to ``ASYNCHRONOUS`` and ``PREALLOCATED_WITH_REALLOC``, and the values set on the XML file are ignored.
```
```suggestion
By default, these values are set to ``ASYNCHRONOUS`` and ``PREALLOCATED_WITH_REALLOC`` within the ``rmw_fastrtps`` implementation, and the values set on the XML file are ignored.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+Among all the configurable attributes, ``rmw_fastrtps`` treats ``publishMode`` and ``historyMemoryPolicy`` differently.
+By default, these values are set to ``ASYNCHRONOUS`` and ``PREALLOCATED_WITH_REALLOC``, and the values set on the XML file are ignored.
+In order to make ``rmw_fastrtps`` use the values in the XML file, the enviranment variable ``RMW_FASTRTPS_USE_QOS_FROM_XML`` must be set to ``1``.
```
```suggestion
In order to use the values in the XML file, the environment variable ``RMW_FASTRTPS_USE_QOS_FROM_XML`` must be set to ``1``.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+By default, these values are set to ``ASYNCHRONOUS`` and ``PREALLOCATED_WITH_REALLOC``, and the values set on the XML file are ignored.
+In order to make ``rmw_fastrtps`` use the values in the XML file, the enviranment variable ``RMW_FASTRTPS_USE_QOS_FROM_XML`` must be set to ``1``.
+
+However, this brings another caveat: If ``RMW_FASTRTPS_USE_QOS_FROM_XML`` is set, but the XML file does not define
```
```suggestion
However, this entails another caveat: If ``RMW_FASTRTPS_USE_QOS_FROM_XML`` is set, but the XML file does not define
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+In order to make ``rmw_fastrtps`` use the values in the XML file, the enviranment variable ``RMW_FASTRTPS_USE_QOS_FROM_XML`` must be set to ``1``.
+
+However, this brings another caveat: If ``RMW_FASTRTPS_USE_QOS_FROM_XML`` is set, but the XML file does not define
+``publishMode`` or ``historyMemoryPolicy``, these attributes take the **FastDDS** default value instead of the ``rmw_fastrtps`` default value.
```
```suggestion
``publishMode`` or ``historyMemoryPolicy``, these attributes take the *Fast DDS* default value instead of the ``rmw_fastrtps`` default value.
```

Previously, italic instead of bold has been used to refer to Fast DDS

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+However, this brings another caveat: If ``RMW_FASTRTPS_USE_QOS_FROM_XML`` is set, but the XML file does not define
+``publishMode`` or ``historyMemoryPolicy``, these attributes take the **FastDDS** default value instead of the ``rmw_fastrtps`` default value.
+For ``historyMemoryPolicy`` this is ``PREALLOCATED``.
+
+``PREALLOCATED`` memory policy does not work with ROS2 topic data types, therefore, in the example we have to explicitly set the ``historyMemoryPolicy`` to a valid value.
```
```suggestion
This is important, specially for ``historyMemoryPolicy``, because the *Fast DDS* deafult value is ``PREALLOCATED`` which does not work with ROS2 topic data types. Therefore, in the example, a valid value for this policy has been explicitly set (``DYNAMIC``).
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+For ``historyMemoryPolicy`` this is ``PREALLOCATED``.
+
+``PREALLOCATED`` memory policy does not work with ROS2 topic data types, therefore, in the example we have to explicitly set the ``historyMemoryPolicy`` to a valid value.
+We also have to especify a default profile for all the topics that ROS2 creates automatically.
```
```suggestion
Bear in mind that this applies to all topics that ROS 2 creates automatically, so at least a default publisher and subscriber profiles setting a valid ``historyMemoryPolicy`` have to be defined.
```

JLBuenoLopez reviewed (commented):
Second partial review (until `Priorization of rmw_qos_profile_t`)

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+ROS 2 RMW only allows for the configuration of certain middleware QoS
+(see `ROS 2 QoS policies <https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings/#qos-policies>`_).
+However, ``rmw_fastrtps`` offers extended configuration capabilities
+to take full advantage of the features in **Fast DDS**.
```
```suggestion
to take full advantage of the features in *Fast DDS*.
```

I do not have a preference between writing Fast DDS in bold or italics, but I think it is better to be consistent throughout the tutorial

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+to take full advantage of the features in **Fast DDS**.
+This tutorial will guide you through a series of examples explaining how to use XML files to unlock this extended configuration.
+
+In order to get more information about using **Fast DDS** on ROS 2, please check the `following documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/ros2.html>`__.
```
```suggestion
In order to get more information about using *Fast DDS* on ROS 2, please check the `following documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/ros2.html>`__.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ROS 2 QoS contained in `rmw_qos_profile_t <http://docs.ros2.org/latest/api/rmw/structrmw__qos__profile__t.html>`_ are always honored, unless set to ``*_SYSTEM_DEFAULT``.
+In that case, XML values (or **Fast DDS** default values in the absences of XML ones) are applied.
```
```suggestion
In that case, XML values (or *Fast DDS* default values in the absence of XML ones) are applied.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+-----------------------------------------
+
+Although we have created a node with two publishers with different configuration, it is not easy to check that they are behaving differently.
+Now that we have covered the basics of XML profiles, let's use them to configure something which has some visual effect on the nodes.
```
```suggestion
Now that the basics of XML profiles have been covered, let us use them to configure something which has some visual effect on the nodes.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+
+Although we have created a node with two publishers with different configuration, it is not easy to check that they are behaving differently.
+Now that we have covered the basics of XML profiles, let's use them to configure something which has some visual effect on the nodes.
+Specifically, we will set a maximum number of matching subscribers on one of the publishers and a partition definition on the other.
```
```suggestion
Specifically, a maximum number of matching subscribers on one of the publishers and a partition definition on the other will be set.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Although we have created a node with two publishers with different configuration, it is not easy to check that they are behaving differently.
+Now that we have covered the basics of XML profiles, let's use them to configure something which has some visual effect on the nodes.
+Specifically, we will set a maximum number of matching subscribers on one of the publishers and a partition definition on the other.
+Note that these are only very simple examples among all the configuration attributes that can be set on ``mrw_fastrtps`` through XML files.
```
```suggestion
Note that these are only very simple examples among all the configuration attributes that can be tuned on ``rmw_fastrtps`` through XML files.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Now that we have covered the basics of XML profiles, let's use them to configure something which has some visual effect on the nodes.
+Specifically, we will set a maximum number of matching subscribers on one of the publishers and a partition definition on the other.
+Note that these are only very simple examples among all the configuration attributes that can be set on ``mrw_fastrtps`` through XML files.
+Please refer to `this documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/xml_configuration.html#xml-profiles>`__ to  see the whole list of attributes that can be configured through XML files.
```
```suggestion
Please refer to `*Fast DDS* documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/xml_configuration.html#xml-profiles>`__ to  see the whole list of attributes that can be configured through XML files.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+Limiting the number of matching subscribers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add the maximum number of matched subscribers to the profile of the ``/async_topic`` publisher. It should look like this:
```
```suggestion
Add a maximum number of matched subscribers to the ``/async_topic`` publisher profile. It should look like this:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+        </matchedSubscribersAllocation>
+    </publisher>
+
+We are limiting the number of matching subscribers to one.
```
```suggestion
The number of matching subscribers is being limited to one.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,646 @@
+    </publisher>
+
+We are limiting the number of matching subscribers to one.
+Now open three terminals. Do not forget to source the setup files and to set the required environment variables.
```
```suggestion

Now open three terminals and do not forget to source the setup files and to set the required environment variables.
```

I suggest beginning a new paragraph

JLBuenoLopez reviewed (commented):
Partial review (only up to `Using partitions within the topic`)

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+Prerequisites
+-------------
+
+This tutorial assumes you have a ROS 2 Foxy (or newer) installation, and that you know how to :ref:`create a package <CreatePkg>`.
```
This tutorial does not work with ROS 2 Foxy. For this tutorial to work with ROS 2 Foxy the ros2/rmw_fastrtps#335 and ros2/rmw_fastrtps#497 should be backported to the foxy branch. Right now, this tutorial only works in ROS 2 Rolling and newer installations.

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+    </publisher>
+
+The number of matching subscribers is being limited to one.
+Now open three terminals and do not forget to source the setup files and to set the required environment variables.
```
```suggestion

Now open three terminals and do not forget to source the setup files and to set the required environment variables.
```

I'd suggest starting a new paragraph with the instructions to run this example

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+The number of matching subscribers is being limited to one.
+Now open three terminals and do not forget to source the setup files and to set the required environment variables.
+On the first terminal run the publisher node, and the subscriber node on the other two.
+You should see that only the first subscriber node is receiving the messages. The second one could not complete the matching process because the publisher prevented it, as it had already reached its maximum of matched publishers.
```
```suggestion
You should see that only the first subscriber node is receiving the messages from both topics.
The second one could not complete the matching process in the ``/async_topic`` because the publisher prevented it, as it had already reached its maximum of matched publishers.
Consequently, only the messages from the ``/sync_topic`` are going to be received in this third terminal:

.. code-block:: console

[INFO] [1613127657.088860890] [sync_async_subscriber]: I heard: 'SYNC: Hello, world! 18'
[INFO] [1613127657.588896594] [sync_async_subscriber]: I heard: 'SYNC: Hello, world! 19'
[INFO] [1613127658.088849401] [sync_async_subscriber]: I heard: 'SYNC: Hello, world! 20'
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+Using partitions within the topic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can use the partitions feature to control which publishers and subscribers exchange information within the same topic.
```
```suggestion
The partitions feature can be used to control which publishers and subscribers exchange information within the same topic.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can use the partitions feature to control which publishers and subscribers exchange information within the same topic.
+Let's change the ``/sync_topic`` publisher to partition ``part1`` while the ``/sync_topic`` subscriber is moved to partition ``part2``.
```
```suggestion
Let us change the ``/sync_topic`` publisher to partition ``part1`` and create a new ``/sync_topic`` subscriber which uses partition ``part2``.
```

JLBuenoLopez reviewed (commented):
Partial review (up to `Configuring a service and client`)

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+* A service subscriber listening to requests on ``/rq/ping``.
+* A service publisher sending responses on ``/rr/ping``.
+* A client publisher sending requests to requests on ``/rq/ping``.
```
```suggestion
* A client publisher sending requests on ``/rq/ping``.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+* A client publisher sending requests to requests on ``/rq/ping``.
+* A client subscriber listening to responses on ``/rr/ping``.
+
+Although you can use these topic names to set the configuration profiles on the XML, sometimes you want to apply the same profile to all services or clients on a node.
```
```suggestion
Although you can use these topic names to set the configuration profiles on the XML, sometimes you may wish to apply the same profile to all services or clients on a node.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+* A client subscriber listening to responses on ``/rr/ping``.
+
+Although you can use these topic names to set the configuration profiles on the XML, sometimes you want to apply the same profile to all services or clients on a node.
+Instead of copying the same profile with all topic names generated for all services, you can just create a publisher and subscriber profile pair named ``service``, and another pair named ``client``.
```
```suggestion
Instead of copying the same profile with all topic names generated for all services, you can just create a publisher and subscriber profile pair named ``service``. The same can be done for clients creating a pair named ``client``.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+Create the nodes with the service and client
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Start creating the node with the service. Add new source file named ``ping_service.cpp`` on your package with the following content:
```
```suggestion
Start creating the node with the service. Add a new source file named ``ping_service.cpp`` on your package with the following content:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+    void ping(const std::shared_ptr<example_interfaces::srv::Trigger::Request> request,
+            std::shared_ptr<example_interfaces::srv::Trigger::Response> response)
+    {
+        response->success = true;
```
```suggestion
        (void)request;
        response->success = true;
```

Avoid unused parameter warning

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+Create the XML profiles for the service and client
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create a file with name ``ping.xml`` and the following content:
```
```suggestion
Create a file with name ``ping.xml`` with the following content:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+
+This configuration file sets the publication mode to ``SYNCHRONOUS`` on the service and to ``ASYNCHRONOUS`` on the client.
+Note that we are only defining the publisher profiles for bot the service and the client, but subscriber profiles could be provided too.
```
```suggestion
Note that we are only defining the publisher profiles for both the service and the client, but subscriber profiles could be provided too.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+^^^^^^^^^^^^^^^^^
+
+Open two terminals and source the setup files on each one.
+Then set the required environment variables for the XML to load:
```
```suggestion
Then set the required environment variables for the XML to be loaded:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+      export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      export RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      export FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      export FASTRTPS_DEFAULT_PROFILES_FILE=path/to/ping.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+      export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      export RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      export FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      export FASTRTPS_DEFAULT_PROFILES_FILE=path/to/ping.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+      SET RMW_IMPLEMENTATION=rmw_fastrtps_cpp
+      SET RMW_FASTRTPS_USE_QOS_FROM_XML=1
+      SET FASTRTPS_DEFAULT_PROFILES_FILE=SyncAsync.xml
```
```suggestion
      SET FASTRTPS_DEFAULT_PROFILES_FILE=path/to/ping.xml
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+
+On the first terminal run the service node.
+You should see the service waiting for requests:
```
```suggestion

.. code-block:: console

    ros2 run sync_async_node_example_cpp ping_service

You should see the service waiting for requests:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+            std::shared_ptr<example_interfaces::srv::Trigger::Response> response)
+    {
+        response->success = true;
+        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Incoming request");
```
```suggestion
        RCLCPP_INFO(rclcpp::get_logger("ping_server"), "Incoming request");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+    {
+        response->success = true;
+        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Incoming request");
+        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "sending back response");
```
```suggestion
        RCLCPP_INFO(rclcpp::get_logger("ping_server"), "Sending back response");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+        rclcpp::Service<example_interfaces::srv::Trigger>::SharedPtr service =
+            node->create_service<example_interfaces::srv::Trigger>("ping", &ping);
+
+        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Ready to serve.");
```
```suggestion
        RCLCPP_INFO(rclcpp::get_logger("ping_server"), "Ready to serve.");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+        while (!client->wait_for_service(1s)) {
+            if (!rclcpp::ok()) {
+            RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Interrupted while waiting for the service. Exiting.");
```
```suggestion
            RCLCPP_ERROR(rclcpp::get_logger("ping_client"), "Interrupted while waiting for the service. Exiting.");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+            RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Interrupted while waiting for the service. Exiting.");
+            return 0;
+            }
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "service not available, waiting again...");
```
```suggestion
            RCLCPP_INFO(rclcpp::get_logger("ping_client"), "Service not available, waiting again...");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "service not available, waiting again...");
+        }
+
+        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Sending request");
```
```suggestion
        RCLCPP_INFO(rclcpp::get_logger("ping_client"), "Sending request");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+        if (rclcpp::spin_until_future_complete(node, result) ==
+            rclcpp::FutureReturnCode::SUCCESS)
+        {
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Response received");
```
```suggestion
            RCLCPP_INFO(rclcpp::get_logger("ping_client"), "Response received");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+        {
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Response received");
+        } else {
+            RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Failed to call service ping");
```
```suggestion
            RCLCPP_ERROR(rclcpp::get_logger("ping_client"), "Failed to call service ping");
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+    [INFO] [1612977403.805799037] [ping_server]: Ready to serve
+    [INFO] [1612977404.807314904] [ping_server]: Incoming request
+    [INFO] [1612977404.836405125] [ping_server]: sending back response
```
```suggestion
    [INFO] [1612977404.836405125] [ping_server]: Sending back response
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+    [INFO] [1612977403.805799037] [ping_server]: Ready to serve
+
+On the second terminal, run the client node.
+You should see the client sending the request and receiving the response:
```
```suggestion

.. code-block:: console

    ros2 run sync_async_node_example_cpp ping_client

You should see the client sending the request and receiving the response:
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,652 @@
+
+.. code-block:: console
+
+    [INFO] [1612977404.805799037] [ping_client]: Sending request
```
```suggestion
    [INFO] [1612977404.805799037] [ping_client]: Sending request
    [INFO] [1613373614.312229535] [ping_client]: Response received
```

JLBuenoLopez reviewed (changes requested):
Review completed.

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,679 @@
+Consequently, only the messages from the ``/sync_topic`` are going to be received in this third terminal:
+
+.. code-block:: console
+[INFO] [1613127657.088860890] [sync_async_subscriber]: I heard: 'SYNC: Hello, world! 18'
```
Insert blank line to prevent warning

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,679 @@
+Note that the synchronous publisher will be publishing on topic ``sync_topic``, while the asynchronous one will be publishing on topic ``async_topic``.
+
+Add a new target on the ``CMakeLists.txt`` file of your package with this code and name it ``SyncAsyncWriter``, so that you can run the node using ``ros2 run``.
+If this node is built and run this node now, both publishers will behave the same, publishing asynchronously in both topics, because this is the default publication mode.
```
```suggestion
If this node is built and run now, both publishers will behave the same, publishing asynchronously in both topics, because this is the default publication mode.
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,679 @@
+``publishMode`` or ``historyMemoryPolicy``, these attributes take the *Fast DDS* default value instead of the ``rmw_fastrtps`` default value.
+This is important, specially for ``historyMemoryPolicy``, because the *Fast DDS* deafult value is ``PREALLOCATED`` which does not work with ROS2 topic data types.
+Therefore, in the example, a valid value for this policy has been explicitly set (``DYNAMIC``).
+
+``PREALLOCATED`` memory policy does not work with ROS2 topic data types, therefore, in the example we have to explicitly set the ``historyMemoryPolicy`` to a valid value.
```
```suggestion
```

JLBuenoLopez commented on source/Tutorials/FastDDS-Configuration/FastDDS-Configuration.rst:
```diff
@@ -0,0 +1,679 @@
+
+Now open three terminals and do not forget to source the setup files and to set the required environment variables.
+On the first terminal run the publisher node, and the subscriber node on the other two.
+You should see that only the first subscriber node is receiving the messages from both topids.
```
```suggestion
You should see that only the first subscriber node is receiving the messages from both topics.
```

JLBuenoLopez reviewed (approved):
Just a warning fix and small typos.

Commit:
add ros2-tsc-charter.rst

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>
(cherry picked from commit baf85ae0ab0c0ff409817e8f6789582f2c4e23fd)

Commit:
fix nitpicks.

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>
(cherry picked from commit 006765cbf14fba646ab54e55384151eafa15072a)

Commit:
fix bold.

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>
(cherry picked from commit 15e1d94c4cc7fa222f1ed5f7468a612876c86614)

Commit:
D005 No newline at end of file

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>
(cherry picked from commit 22a8c353a69ee26bb525011f6e1e6420dcd44e05)

Commit:
fix typo

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>
(cherry picked from commit 5fc458b351a3c0ad40ea44208503f5e899277b35)

Commit:
one sentence per line.

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>
(cherry picked from commit 33c1088aa12c2444090ec1900cdfdd2ca06bbd6b)

clalancette committed:
Remove the PDF version, move links to the html version.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit bc6c0876c043f37c7b56458c37afde36192e0a75)

clalancette committed:
Rename the charter file to ROS2-TSC-Charter.rst.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 92f6e0b5ca976e75493a58cb4c29e4b9c34256f5)

clalancette committed:
Remove the executable bit from sony.png.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit ff0de062e8923fc05fdc89e17256acf9993ab798)

clalancette committed:
Add the Governance to the toctree.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit 5ddf7888d30dd0b309f3815aa1637d9c238f9190)

clalancette committed:
Collapse the headings down into one.

That way it renders more nicely from other pages.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
(cherry picked from commit bcd1972390b551b93255772023e245a375362178)

mergify[bot] committed:
Add documentation for using Python packages (#966) (#1106)

* Add documentation for using Python packages

Signed-off-by: Audrow Nash <audrow.nash@gmail.com>
Co-authored-by: Michael Carroll <michael@openrobotics.org>

clalancette committed:
Add ros2 param load documentation to the rolling branch. (#1105)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

maryaB-osr committed:
create guides page

Signed-off-by: maryaB-osr <marya@openrobotics.org>

add/update doc-type descriptions
also update some other small parts of the overview page

Signed-off-by: maryaB-osr <marya@openrobotics.org>

partial move

Signed-off-by: maryaB-osr <marya@openrobotics.org>

partial move

Signed-off-by: maryaB-osr <marya@openrobotics.org>

move troubleshooting to guides, add redirect-from to pages moved so far

Signed-off-by: maryaB-osr <marya@openrobotics.org>

moved remaining pages, added redirects

Signed-off-by: maryaB-osr <marya@openrobotics.org>

clalancette committed:
Add crystal to the list of branches to be built by multiversion. (#1118)

* Add crystal to the list of branches to be built by multiversion.

Also make sure to mark it EOL.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

mergify[bot] committed:
Fix URLs to ros2.repos files (#1119) (#1121)

'release-foxy' does not point to a valid file and 'release-latest' is no longer used.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>
(cherry picked from commit b0f7676456e78b6ff68cc80b8bbaf5a12cb94372)

# Conflicts:
#	source/Installation/Maintaining-a-Source-Checkout.rst

Co-authored-by: Jacob Perron <jacob@openrobotics.org>

mergify[bot] committed:
Record Tier IV's decision to depart the TSC (#1127) (#1128)

Co-authored-by: Brian Gerkey <gerkey@openrobotics.org>
(cherry picked from commit 4d34976aa3667f43a2edb00a666f76b2e7f14299)

Co-authored-by: gerkey <gerkey@osrfoundation.org>

mergify[bot] committed:
Fix typo in Launch-files-migration-guide.rst (#1140) (#1141)

(cherry picked from commit 1a5d5b2ded832068256b47031b31351f654d9b49)

Co-authored-by: Connor Anderson <connoranderson@users.noreply.github.com>

mergify[bot] committed:
Add config and launch directories to recommended package layout (#1139) (#1144)

These two directories seem common enough, and already a convention,
that we should consider adding them to the recommended package layout.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>
(cherry picked from commit 5eb270f9ef34f846aaa728ee8ec973a49654921d)

Co-authored-by: Jacob Perron <jacob@openrobotics.org>

maryaB-osr committed:
Remove installation landing page [rolling] (#1132)

* remove rolling installation landing page

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* remove rolling installation landing page

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* Fix the links.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

* shorten title, mention distro in first line

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* shorten title

Signed-off-by: maryaB-osr <marya@openrobotics.org>

* Fix one more installation link.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

Co-authored-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Update the README.md file to explain how to contribute. (#1147)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

razr committed:
update Wind River logo (#1152)

maryaB-osr committed:
fix reference links

Signed-off-by: maryaB-osr <marya@openrobotics.org>

maryaB-osr committed:
fix reference link typos

Signed-off-by: maryaB-osr <marya@openrobotics.org>

maryaB-osr committed:
return deleted redirects

Signed-off-by: maryaB-osr <marya@openrobotics.org>

clalancette committed:
Add the mergify rules to the 'rolling' branch. (#1155)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

razr committed:
Update Real-time Leads (#1159)

clalancette committed:
Enable RTD theme. (#1150)

The read-the-docs theme is far nicer (in my opinion) than the
default 'alabaster' theme.  Also, it gives us links to GitHub
sources and the version picker without additional plugins
needed.  This theme also puts the table-of-contents for
subsections in the order they were written, not in alphabetical
order.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

christophebedard committed:
Remove galactic note in logging tutorial (#1164)

Signed-off-by: Christophe Bedard <bedard.christophe@gmail.com>

clalancette committed:
Add in distribution name to title. (#1163)

This should make it a little easier for readers to understand
which branch of the documentation they are reading.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

maryaB-osr committed:
apply review comments

Signed-off-by: maryaB-osr <marya@openrobotics.org>

IkerLuengo committed:
removing note about working only on rolling

Since the PR was retargeted to the rolling branch, the remark is not
needed anymore

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

IkerLuengo committed:
Review requests: explain code and concepts, correct typos

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

IkerLuengo committed:
review requests: typos and CMakeLists examples

Signed-off-by: Iker Luengo <ikerluengo@eprosima.com>

clalancette committed:
Fix typo.
